### ISSUE TITLE
i cant login Crypto Wallets

### ISSUE THREAD

**Issue by tharinduvindula** [brave/brave-browser#6442]
## Steps to Reproduce

   1. i can't log brave Crypto Wallet
   2. i can't Restore account
   3. i can't do anything in brave Crypto Wallet, only show that interface

## Actual result:

![Capture](https://user-images.githubusercontent.com/43117293/66676154-75573b00-ec84-11e9-9f9f-85c7edf82847.JPG)

## Version/Channel Information:

- Can you reproduce this issue with the current release?
- Can you reproduce this issue with the beta channel?
- Can you reproduce this issue with the dev channel?
- Can you reproduce this issue with the nightly channel?

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields?
- Does the issue resolve itself when disabling Brave Rewards?
- Is the issue reproducible on the latest version of Chrome?

**Comment by srirambv** [comment]
Can't reproduce the issue. Can you please add the browser details and check brave://components for `Crypto Wallets` version?

**Comment by ivigamberdiev** [comment]
I have same issue. My case:
1) I created wallet and wrote password
2) But accidentally closed the tab with wallet w/o entering a seed-phrase
3) When try login or restore nothing happening
I checked Metamask for this bug, but in this test case it open tab with entering a seed-phrase. Looks like Brave specific bug.

**Comment by tharinduvindula** [comment]
I type my correct password but it after log show dashboard and after few seconds it reroute to login page

**Comment by srirambv** [comment]
Multiple +1 from https://www.reddit.com/r/brave_browser/comments/djigbd/the_wallet_is_awful_please_fix/

**Comment by arosboro** [comment]
I tried importing my metamask 12 word mnemonic phrase, but got the response that 12 words was not long enough (16, or 24 I believe).  I quit, came back to this.  When I enter the password I remember, I get:  `MetamaskController - No HD Key Tree found`

**Comment by srirambv** [comment]
Could you guys check what component version is in the browser? You can check the component version from brave://component. The current stable release is `1.0.11`. Please click on `Check for update` if your browser is having an older version. Once updated restart the browser and check if you are able to login

**Comment by arosboro** [comment]
@srirambv I don't have a path at brave://component, but brave://version says:

Brave | 0.69.135 Chromium: 77.0.3865.120 (Official Build) (64-bit)
-- | --
Revision | 416d6d8013e9adb6dd33b0c12e7614ff403d1a94-refs/branch-heads/3865@{#884}

**Comment by arosboro** [comment]
Nevermind, found it at brave://components.
Crypto Wallets - Version: 1.0.11
Status - Up-to-date

**Comment by tharinduvindula** [comment]
my brave://wallet/  is not work for me  but i can use meta mask

**Comment by ivigamberdiev** [comment]
This problem is actual, even after update to Brave Version 0.70.121. Pls reopen it.

**Comment by srirambv** [comment]
Reopening the issue as this needs investigation as to why the login is failing.

**Comment by arosboro** [comment]
Can someone from the team at least provide instruction on how to wipe the crypto wallets component data/seed and start from scratch?

**Comment by Brave-Matt** [comment]
+1
https://community.brave.com/t/cant-log-into-cryptowallet/90681

**Comment by pcatana** [comment]
Got the same issue after updating to the latest version
Version 1.0.0 Chromium: 78.0.3904.97 (Official Build) (64-bit)

**Comment by Brave-Matt** [comment]
Another on from Community:
https://community.brave.com/t/re-crypto-wallet-blank-page/85967/8

@srirambv what's the status on this?

**Comment by DaLN** [comment]
> I have same issue. My case:
> 
> 1. I created wallet and wrote password
> 2. But accidentally closed the tab with wallet w/o entering a seed-phrase
> 3. When try login or restore nothing happening
>    I checked Metamask for this bug, but in this test case it open tab with entering a seed-phrase. Looks like Brave specific bug.
> 
> Brave	0.69.135 Chromium: 77.0.3865.120 (Official Build) (64-bit)
> Revision	416d6d8013e9adb6dd33b0c12e7614ff403d1a94-refs/branch-heads/3865@{#884}
> Crypto Wallets - Version: 1.0.10

Similar issue here, when creating the wallet, at the verification phase i could not choose the seed phrase as it was blank, closed the window, then can't even reset or reimport anything.

**Comment by srirambv** [comment]
Could someone having login issues try this

1. Login with the correct password
2. When it says its incorrect
3. Close and relaunch the browser
4. Try logging again with correct password and check if it works

**Comment by aaron-hanson** [comment]
> Could someone having login issues try this
> 
> 1. Login with the correct password
> 2. When it says its incorrect
> 3. Close and relaunch the browser
> 4. Try logging again with correct password and check if it works

Tried that, no luck.  Also, is there some way to reset things completely to try and set up from scratch with a new password, so we can at least get around this?

**Comment by srirambv** [comment]
@aaron-hanson just to be sure the crypto wallet component is up to date right? Could you check the version in `brave://components`? The current released version is `1.0.13`

> is there some way to reset things completely 

Should be done as part of #5741

**Comment by aaron-hanson** [comment]
> @aaron-hanson just to be sure the crypto wallet component is up to date right? Could you check the version in `brave://components`? The current released version is `1.0.13`
> 
> > is there some way to reset things completely
> 
> Should be done as part of #5741

Yep, it shows`Crypto Wallets - Version: 1.0.13`

**Comment by Brave-Matt** [comment]
Another +1 from Reddit:
https://www.reddit.com/r/brave_browser/comments/e5xj3y/why_there_isnt_technical_support_from_team_for/

**Comment by Brave-Matt** [comment]
Several reports on Reddit:
https://community.brave.com/t/cypto-wallet-password-and-recovery/93629/10

**Comment by arosboro** [comment]
I have a reasonable workaround to this after reading the comments in that thread @Brave-Matt .

I am on a mac, but pc users may be able to find the brave application profile in the mentioned thread, it's in the Application Data folder I believe.  The browser specific steps will work on either platform.  In reproducing the bug, I found that it is either intermittently successful at importing metamask wallets (I got it second try, after using a different password), or it's related to the password you choose.

Make sure you do these steps before attempting to fix the crypto wallet issue
1. Have sync enabled and verify that you can add a computer from your other device (It should show a 24 word secure phrase.
2. Store/Print/Write down your brave rewards wallet words.

Here are the steps I took to successfully import my metamask wallet.
1. Quit Brave Browser
2. Rename `~/Library/Application Support/BraveSoftware` to back it up (It's your application profile)
3. Open Brave Browser
4. Open Crypto wallets
5. Select the option to restore your account/wallet with wallet seed phrase
6. Type an entirely new passphrase that you have not used with metamask before
7. Accounts load and tokens at your address are displayed correctly
8. Restore Brave rewards wallet seed
9. Connect to sync and get your bookmarks back

Here are the steps I took to reproduce the bug:
1. Quit Brave Browser
2. Rename `~/Library/Application Support/BraveSoftware` to back it up (It's your application profile)
3. Open Brave Browser
4. Skip Introduction onboard tour
5. Open crypto wallets and choose restore wallet option
6. Restore 12 word metamask phrase from metamask chrome store extension
7. Enter the same password as the chrome store extension for the passphrase
8. Brave Crypto wallet ui spinner appears and does not respond
9. Close and re-open the browser, come back to crypto wallet and this issue occurs. I can't log in/re-import seed phrase.

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.2.19 Chromium: 79.0.3945.56 (Official Build) beta (64-bit)
-- | --
Revision | 73cc6bf591f792b99f8fc7cdfb8addedbd084bf8-refs/branch-heads/3945@{#788}
OS | Linux
Component  | `0.1.30`

New Wallet
* Verified multiple failed login attempts and then with correct password still allows you to login
* Verified console doesn't log any errors when the wrong password is used
* Verified no browser restart is required to login using correct password

Import wallet
* Verified importing Brave Crypto Wallet seed and multiple failed login attempts and then with correct password still allows you to login
* Verified importing MM seed and multiple failed login attempts and then with correct password still allows you to login

**Comment by aeikenberry** [comment]
This is still happening for me. Entering the correct password flashes something is loading, and then takes me back to login, entering an incorrect password says "incorrect password".

**Comment by ryanml** [comment]
@aeikenberry - can you navigate to brave://inspect#extensions and inspect Crypto Wallets, try to log in again, and let me know what you see in the console tab?

**Comment by aeikenberry** [comment]
<img width="1160" alt="Screen Shot 2020-01-19 at 10 24 44 AM" src="https://user-images.githubusercontent.com/599112/72684436-0751b980-3aa6-11ea-9da4-fd4388efc2cf.png">
The `Invalid asm.js` warning comes up when you open the crypto wallets window

**Comment by aeikenberry** [comment]
I think I somehow messed up my restore phrase stuff setting up this wallet. Is there a way to wipe the wallet and setup a new local one?

**Comment by aeikenberry** [comment]
When i click Restore account? nothing happens in the UI, this warning in the console:
```
MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 "update" listeners added. Use emitter.setMaxListeners() to increase limit.
```

**Comment by aeikenberry** [comment]
Removed the ApplicationSupport directory and started again, didn't have the same issue.

**Comment by aeikenberry** [comment]
I was able to reproduce the issue on Brave Beta.

1. Create a new local wallet with a new password.
2. On the backup pass screen, exit out of it without continuing.
3. Go back to the wallet page, see the login page, new password doesn't work.

**Comment by BadseyBurger** [comment]
I am also having this problem. The workaround further up the thread doesn't work for me. Help please :)

**Comment by foiba** [comment]
Exact same problem for almost a month now. Nothing helps. I log in with password, screen flashes showing a dropdown for about .3s in upper right corner and immediately redirects to login. I tried typing a random password and that shows as incorrect password. So it definitely recognizes my actual password as correct. No workarounds help so far. Great browser, hoping for a patch or a workaround that works around!

**Comment by denisediniz** [comment]
Same here. Steps to reproduce the issue (everything up-to-date)

1. Create a new local wallet with a new password;
2. Close the tab of the seed phrase confirmation page;
3. Go back to the crypto-wallets page: the password immediately redirects to login and there's no way to restore, remove or create a new account.

the seed phrase confirmation page doesn't scroll down in Win10 desktop.

**Comment by ryanml** [comment]
Thank you all for your reports, it’s helpful as I try to see exactly what’s going on. @BadseyBurger @foiba - can you confirm if you backed out of the seed confirmation page like @denisediniz and @aeikenberry ?

**Comment by DruPlusDru** [comment]
I am also having this same problem. The seed confirmation page froze up so I was unable to scroll the page and complete the assigned task. I closed out the window and am having the same issue as @denisediniz and @aeikenberry

**Comment by s-ben** [comment]
If it's any help, was locked out of MetaMask for a web3 site I was trying to access (Gitcoin), even after disabling Brave Crypto Wallets and restarting Brave. Restarted computer (OSX 10.14.6) and only then did MetaMask successfully connect.

**Comment by dhniels** [comment]
I'm having this problem as well. The seed confirmation screen only showed 4 words and I couldnt get past it so i closed the window and now i get the login screen. If i enter the wallet password I created, it flashes the wallet very briefly before going back to the same login screen. Can't get past this screen or do really anything at all now.

**Comment by ryanml** [comment]
@BadseyBurger @foiba @denisediniz @aeikenberry @DruPlusDru @s-ben @dhniels 

Thank you all so much for your patient, I have identified a fix for this and posted it here: https://github.com/brave/ethereum-remote-client/pull/114

We have also fixed the scroll issue on the confirm seed phrase page. You'll now be able to see everything on certain viewports. The fix will also backfix your situations so login will work as expected, no resets needed. This fix is set go out by the end of this week.

Here is a video of the fix (sluggishness is due to testing on a debug build):
https://ryanlane.se/img/login-fix.gif

**Comment by ryanml** [comment]
@BadseyBurger @foiba @denisediniz @aeikenberry @DruPlusDru @s-ben @dhniels - please see https://www.reddit.com/r/brave_browser/comments/f2uqsv/crypto_wallet_is_one_giant_bug/fhet7nf/ regarding the update. It would be great if you guys could check it out and attempt to log in.

**Comment by foiba** [comment]
@ryanml I tried refreshing the cryptowallet tab and opening a new one and it failed to work however i was able to log in when I restarted Brave. It doesn't show the BAT I've accumulated and been rewarded but that's something else. Thank you!

**Comment by denisediniz** [comment]
Thank you! The Restore account links redirects to the right page (the Restore page isn't scrollable yet). I can't login right now but just because I made the workaround to delete my local test wallet and I can't create a new wallet, so right now I can't test the login function. I'm going to follow the #7677 issue.

**Comment by ryanml** [comment]
@denisediniz Are you not able to submit the account restore form due to a scrolling issue? Can I see a screenshot of this? Do you have your restore phrase that you can use?

**Comment by denisediniz** [comment]
Unfortunately I don't have the restore phrase for the deleted wallet. I had removed my local testing wallet through the workaround of deleting the odb... folders within User Data to create another one. As I'll need to create another account to login, with new password and restore phrase, I'm watching the other issue. The Restore acc page isn't scrollable but it works well through tabbing into the fields. Ty =)

**Comment by ryanml** [comment]
@denisediniz Got it, I haven't heard any other reports of scrolling issues on the Restore page but I will file an issue for it. I'd like to help you get back and running on Crypto Wallets, can you email me at rlanese at brave.com so I can work with you? Thanks!

**Comment by HikariWS** [comment]
I'm having similar issue. I just installed Brave a few days ago, so everything's updated.

I installed MetaMask and went to connect my Trezor to it. It was a very easy process on Chrome, but on Brave I keep on loading message on https://connect.trezor.io/7/popup.html#loading

Same thing happens for chrome://wallet/ when I try to connect my Trezor.

Oddly, MEW works and is able to connect.

As it always worked on Chrome flawlessly, my bet is the issue is on Brave. I wanna move out of Chrome because it bypasses pihole and keeps showing banners, it's sad Brave isn't working with Trezor.
